# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../exact-image/hotfix.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Index: edisplay.hh
===================================================================
--- ./edisplay/edisplay.hh	(revision 1193)
+++ ./edisplay/edisplay.hh	(revision 1194)
@@ -8,6 +8,9 @@
     image = new Image;
   }
   ~Viewer() {
+    if (evas_data) {
+      delete (evas_data); evas_data = 0;
+    }
     delete (image); image = 0;
   }
   
Index: edisplay.cc
===================================================================
--- ./edisplay/edisplay.cc	(revision 1193)
+++ ./edisplay/edisplay.cc	(revision 1194)
@@ -285,7 +285,7 @@
   evas->FontPathPrepend ("/usr/X11/lib/X11/fonts/TrueType/");
   evas->FontPathPrepend ("/opt/e17/share/evas/data/");
   evas->FontPathPrepend ("/usr/X11/share/fonts/TTF/");
-  
+ 
   if (true) {
     evas->ImageCache (1024 * 1024);
     evas->FontCache (256 * 1024);
@@ -575,13 +575,14 @@
     }
   
   if (evas_image) {
-    free (evas_image->Data());
     delete evas_image;
     evas_image = 0;
   }
   
-  delete evas_bgr_image;
-  evas_bgr_image = 0;
+  delete evas_osd_text1; evas_osd_text1 = 0;
+  delete evas_osd_text2; evas_osd_text2 = 0;
+  delete evas_osd_rect;  evas_osd_rect = 0;
+  delete evas_bgr_image; evas_bgr_image = 0;
   
   delete evas;
   evas = 0;
